MIGRATING_v2_to_v3.md

```diff
@@ -0,0 +1,223 @@
+# Okta Java Management SDK Migration Guide
+ 
+This SDK uses semantic versioning and follows Okta's 
+[library version policy](https://developer.okta.com/code/library-versions/). 
+In short, we do not make breaking changes unless the major version changes!
+
+## Migrating from 2.x.x to 3.0.0
+
+Version 3.0.0 of this SDK introduces a number of breaking changes from previous versions. 
+In addition to many new classes/interfaces, some existing classes/interfaces are no longer backward compatible due to method renaming and signature changes.
+
+### Package `com.okta.sdk.client.Client`
+
+Below method has been added.
+   - `void deleteIdentityProviderKey(String keyId)`
+   
+Below methods have undergone a signature change.
+   - `listGroups(String q, String filter)` signature changed to `listGroups(String q, String filter, String expand)`
+   - `listPolicies(String s)` return type changed from `com.okta.sdk.resource.policy.PolicyList` to `com.okta.sdk.resource.authorization.server.AuthorizationServerPolicyList`
+
+### Package `com.okta.sdk.resource.application.AcsEndpoint`
+
+This is a newly created interface with methods listed below.
+   - `Integer getIndex()`
+   - `AcsEndpoint setIndex(Integer index)`
+   - `String getUrl()`
+   - `AcsEndpoint setUrl(String url)`
+
+### Package `com.okta.sdk.resource.application.Application` 
+
+Below method has been added.
+   - Added `ApplicationGroupAssignment createApplicationGroupAssignment(String groupId)` method.
+   This method allows assigning a group to an Application.
+
+### Package `com.okta.sdk.resource.application.OpenIdConnectApplicationSettingsClient`
+
+Below methods have been added.
+   - `OpenIdConnectApplicationSettingsRefreshToken getRefreshToken()`
+   - `OpenIdConnectApplicationSettingsRefreshToken setRefreshToken(OpenIdConnectApplicationSettingsRefreshToken refreshToken)`
+   - `OpenIdConnectApplicationSettingsClientKeys getJwks()`
+   - `OpenIdConnectApplicationSettingsClient setJwks(OpenIdConnectApplicationSettingsClientKeys jwks)`
+   
+### Package `com.okta.sdk.resource.application.OpenIdConnectApplicationSettingsClientKeys`
+
+This is a newly created interface with methods listed below.
+   - `JsonWebKeyList getKeys()`
+   - `OpenIdConnectApplicationSettingsClientKeys setKeys(JsonWebKeyList keys)`
+
+### Package `com.okta.sdk.resource.application.OpenIdConnectApplicationSettingsRefreshToken`
+
+This is a newly created interface with methods listed below.
+   - `Integer getLeeway()`
+   - `OpenIdConnectRefreshTokenRotationType getRotationType()`
+   - `OpenIdConnectApplicationSettingsRefreshToken setLeeway(Integer leeway)`
+   - `OpenIdConnectApplicationSettingsRefreshToken setRotationType(OpenIdConnectRefreshTokenRotationType rotationType)`
```

Comment:
I don't see a section that lists this new type `OpenIdConnectRefreshTokenRotationType`.

Reply:
updated